### PR TITLE
Implement fast insert for mysql and sqlite

### PR DESCRIPTION
**Pull request by dstandish**
We presently have a "fast path" for postgres.  But we can use similar logic for mysql and sqlite.

### CODE REVIEW

File: airflow/datasets/manager.py [hunk 1]
```diff
@@ -94,14 +95,26 @@ def _slow_path_queue_dagruns(self, dataset: DatasetModel, session: Session) -> N
...
+        elif session.bind.dialect.name == "sqlite":
+            from sqlalchemy.dialects.sqlite import insert
+        elif session.bind.dialect.name == "mysql":
+            from sqlalchemy.dialects.mysql import insert
+
+            prefix = 'IGNORE'
+        else:
+            raise ValueError("Only sqlite, postgres, and mysql supported with this method.")
+        params = [{'target_dag_id': target_dag.dag_id} for target_dag in dataset.consuming_dags]
+        if params:
```
Comment (jedcunningham) [inline airflow/datasets/manager.py:111]: Would this ever be false?

Comment (uranusjr) [reply airflow/datasets/manager.py:111]: It would if `consuming_dags` is empty (not sure if it’s possible in the context).

Comment (jedcunningham) [reply airflow/datasets/manager.py:111]: Yeah, and that's checked in `register_dataset_change` before we get here.

Comment (dstandish) [reply airflow/datasets/manager.py:111]: yeah... this is vestigial... a remnant from refactor.... 
it doesn't hurt to be defensive in this way, and handle calling with empty set... but better readability would be to exit based on empty dag_ids list!

**Comment by potiuk** [comment]
I think it would be great to refer to some docs why this is better/faster? I looked in the docs and could not easily find it and reason about it ?

**Comment by ashb** [comment]
@potiuk This is faster as it is running one query with multiple values with (the db-specific equivalent of) `ON CONFLICT DO NOTHING`, where as the slow path takes a save point before each row, tries to insert the row, and on exception rolls back to the save point.

So fast path is faster because it's one query, not n, and doesn't involve exceptions (py) or savepoints (db) side!

@dstandish Can you convert ^^ to a code comment in appropriate place?

**Comment by dstandish** [comment]
> @potiuk This is faster as it is running one query with multiple values with (the db-specific equivalent of) `ON CONFLICT DO NOTHING`, where as the slow path takes a save point before each row, tries to insert the row, and on exception rolls back to the save point.
> 
> So fast path is faster because it's one query, not n, and doesn't involve exceptions (py) or savepoints (db) side!
> 
> @dstandish Can you convert ^^ to a code comment in appropriate place?

sure thing, just holding off on merging this one until event timestamp one goes in, then will have to rebase and resolve conflicts

**Pull request closed without merging**